toolchain/parse/handle_decl_name_and_params.cpp

```diff
@@ -49,12 +50,14 @@ auto HandleDeclNameAndParams(Context& context) -> void {
   if (auto identifier = context.ConsumeIf(Lex::TokenKind::Identifier)) {
     HandleName(context, state, *identifier,
                NodeKind::IdentifierNameNotBeforeParams,
+               NodeKind::IdentifierNameQualifierWithoutParams,
                NodeKind::IdentifierNameBeforeParams);
     return;
   }
 
   if (auto keyword = context.ConsumeIf(Lex::TokenKind::Destroy)) {
     HandleName(context, state, *keyword, NodeKind::KeywordNameNotBeforeParams,
+               NodeKind::KeywordNameQualifierWithoutParams,
```

Comment:
We could potentially reject this case in the parse stage.

Reply:
True, but I was leaning towards parity with parsing syntax here. Also, much as `fn destroy {}` should be invalid, so should `fn destroy() {}`, and it would be nice to have that diagnostic emitted consistently since the issue could be identified the same way (must be a method).

Reply:
Extended note, I mainly value parsing parity because there's the discussion about having more of these keyword functions, and so a generic parsing approach may lend itself best towards future choices. Also, having more "looks like this structure" could help with diagnostics.